static/js/services/select2-search.js

```diff
@@ -155,8 +155,27 @@ angular.module('inboxServices').factory('Select2Search',
             });
           selectEl.after(button);
         }
+
+        // Hydrate and re-set real doc on change
+        // !tags -> only support single values, until there is a use-case
+        if (!tags) {
+          selectEl.on('select2:select', function(e) {
+            var docId = e.params &&
+                        e.params.data &&
+                        e.params.data.id;
+
+            if (docId) {
+              DB().get(docId).then(function(doc) {
+                selectEl.select2('data')[0].doc = doc;
+
+                selectEl.trigger('change');
+              });
```

Comment:
I'm just wondering what to do if there's an error (say if an admin goes offline). Maybe just catch and call $log.error with a useful message?